content-services/content-sources-backend
# Fixes 2930: API support for list / search environments

## Summary

This adds:

- `repositories/{uuid}/environments` endpoint for listing, returns ID, Name, and Description metadata
- `environments/names` endpoint for searching, returns Name and Description of all environments found across the repositories requested
- database support to store environment metadata
- environments are de-duplicated based on combination of name/ID

## Testing steps

Endpoint: `$BASE_URL/environments/names`

Example request: `{ "urls": [ "https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/" ], "search": "kde" }`

- add a repository or 2 that has a comps.xml file (e.g. https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/) and make a request to `repositories/{uuid}/environments`, should list environments associated with the repo UUID
- make a request to `environments/names` with at least one valid repo URL or repo UUID and a search term, should return the environments with that search term in the name

## Checklist
